gqcpy/src/QCMethod/CC/CCSDEnvironment_bindings.cpp

```diff
@@ -28,6 +28,30 @@ namespace py = pybind11;
 namespace gqcpy {
 
 
+/**
+ *  Convert a NumPy array to a rank-four Eigen::Tensor since the automatic conversion from a NumPy array 
+ *  to an Eigen::Tensor is not supported yet.
+ * 
+ *  @param array       the NumPy array that should be converted to a tensor
+ * 
+ *  @return the corresponding rank-four Eigen::Tensor
+ */
+template <typename T>
+Eigen::Tensor<T, 4> asTensor(const py::array_t<T>& inArray) {
+    // based on https://github.com/pybind/pybind11/issues/1377
+    // request a buffer descriptor from Python
+    py::buffer_info buffer_info = inArray.request();
+
+    // extract data an shape of input array
+    T *data = static_cast<T *>(buffer_info.ptr);
+    std::vector<ssize_t> shape = buffer_info.shape;
+
+    // wrap ndarray in Eigen::Map
+    Eigen::Tensor<T, 4>  in_tensor = Eigen::TensorMap<Eigen::Tensor<T, 4>>(data, shape[0], shape[1], shape[2], shape[3]);
+    return in_tensor;
+}
```

Comment:
Could you clean up this source code?
- The line comments don't start with a capital
- Sometimes, there are too many white spaces (use `clang-format` for this)
- There's no need to manually line break line 32
- You can use `const auto` in many cases.